app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -393,6 +401,80 @@ class HomeActivityTest {
     }
   }
 
+  // TODO(#2057): Remove when TextViews are properly measured in Robolectric.
+  @RunOn(TestPlatform.ESPRESSO) // Incorrectly passes on Robolectric and shouldn't be re-enabled
+  @Config(qualifiers = "port-mdpi")
+  @Test
+  fun testHomeActivity_phonePort_longProfileName_welcomeMessageIsDisplayed() {
+    launch<HomeActivity>(createHomeActivityIntent(longNameInternalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(0)
+      onView(
+        atPositionOnView(
+          R.id.home_recycler_view,
+          0,
+          R.id.profile_name_textview
+        )
+      ).check(matches(not(isEllipsized())))
+    }
+  }
+
+  // TODO(#2057): Remove when TextViews are properly measured in Robolectric.
+  @RunOn(TestPlatform.ESPRESSO) // Incorrectly passes on Robolectric and shouldn't be re-enabled
+  @Config(qualifiers = "land-mdpi")
+  @Test
+  fun testHomeActivity_phone_configChange_longProfileName_welcomeMessageIsDisplayed() {
+    launch<HomeActivity>(createHomeActivityIntent(longNameInternalProfileId)).use {
```

Comment:
@anandwana001 We are using `longNameInternalProfileId` in the new test cases

Reply:
any specific reason for this?

Reply:
Yes, because we want to check that if the name is long, then is welcome_message is getting displayed correctly or not. For that, we want to launch the HomeActivity with the profiled of the long name.